framework/audio/engine/internal/audioengine.cpp

```diff
@@ -167,12 +228,76 @@ void AudioEngine::setPlaySpeed(handle h, float speed)
     m_sl->engine.setRelativePlaySpeed(h, speed);
 }
 
-mu::async::Notification AudioEngine::playCallbackCalled() const
+void AudioEngine::swapPlayContext(handle h, Context& ctx)
 {
-    return m_playCallbackCalled;
+    HandleMeta& m = meta(h);
+    IF_ASSERT_FAILED(m.isValid()) {
+        return;
+    }
+
+    m.playContext.swap(ctx);
+    if (m.playContextRequested) {
+        m.playContextChanged.send(m.playContext);
+    }
+
+    if (m.playContext.hasVal(CtxKey::InstanceDestroyed)) {
+        onStop(h);
+    }
 }
 
-void AudioEngine::onPlayCallbackCalled()
+mu::async::Channel<Context> AudioEngine::playContextChanged(handle h) const
 {
-    m_playCallbackCalled.notify();
+    HandleMeta& m = const_cast<AudioEngine*>(this)->meta(h);
+    IF_ASSERT_FAILED(m.isValid()) {
+        return mu::async::Channel<Context>();
+    }
+    m.playContextRequested = true;
```

Comment:
Where is the false flag set for it?

Reply:
Nowhere :)
This flag means that we have been getting change the context channel, and, accordingly, we must send data to it.
We can always send data to the channel, but it's quite expensive, I don't want to do it just like that